src/Npgsql/NpgsqlBinaryExporter.cs

```diff
@@ -388,9 +388,8 @@ async ValueTask DisposeAsync(bool async)
                 Expect<ReadyForQueryMessage>(await _connector.ReadMessage(async, cancellationToken: default), _connector);
             }
 
-            var connector = _connector;
+            _connector.EndUserAction();
```

Comment:
IIRC the intention here was to provide better guarding against (incorrect) concurrent use of (non-multiplexed) connections by the user, by calling EndUserAction as late as possible (this is what releases the user lock). But since Cleanup releases multiplexed connections back to the pool, it's obvious that we should call EndUserAction after that.

We should rethink the general concurrency detection strategy - keeping both multiplexing and non-multiplexing in mind. I've opened #3255 to track this, but I think that's definitely not for 5.0. For now I'm good with this change.

(a Rust-like language ownership model would have been very useful here :/)